apps/src/codebridge/FilePreview/HTMLPreview.tsx

```diff
@@ -34,17 +36,48 @@ export const HTMLPreview = () => {
   const [debouncedSource, setDebouncedSource] = useState(source);
   const sourceLevelId = useRef<number | undefined>(undefined);
   const [isLevelLoading, setIsLevelLoading] = useState(false);
-  const [currentFile, setCurrentFile] = useState<string>('index.html');
+  const [currentFile, setCurrentFile] = useState<string>(
+    DEFAULT_START_HTML_FILE
+  );
   const isPredictLevel = levelProperties?.predictSettings?.isPredictLevel;
   const hasSubmittedPredictResponse = useAppSelector(
     isPredictResponseSubmitted
   );
   const allowUserScripts = !isPredictLevel || hasSubmittedPredictResponse;
+  const canNavigateBack = navigationHistoryIndex > 0;
+  const canNavigateForward =
+    navigationHistoryIndex < navigationHistory.length - 1;
+
+  const onNavigateBack = () => {
+    if (!canNavigateBack) {
+      return;
+    }
+    const updatedFile = navigationHistory[navigationHistoryIndex - 1];
+    setNavigationHistoryIndex(navigationHistoryIndex - 1);
+    setCurrentFile(updatedFile);
+    iframeRef.current?.contentWindow?.postMessage(
+      {type: IframeMessageType.NAVIGATE_TO_FILE, fileName: updatedFile},
+      previewUrl
+    );
+  };
+  const onNavigateForward = () => {
+    if (!canNavigateForward) {
+      return;
+    }
+    const updatedFile = navigationHistory[navigationHistoryIndex + 1];
+    setNavigationHistoryIndex(navigationHistoryIndex + 1);
+    setCurrentFile(updatedFile);
+    iframeRef.current?.contentWindow?.postMessage(
+      {type: IframeMessageType.NAVIGATE_TO_FILE, fileName: updatedFile},
+      previewUrl
+    );
+  };
 
   useLifecycleNotifier(LifecycleEvent.LevelLoadStarted, () => {
     // When we switch levels, clear the source so the preview does not show outdated content.
     setDebouncedSource(undefined);
     setIsLevelLoading(true);
+    setCurrentFile(DEFAULT_START_HTML_FILE);
```

Comment:
do we also want to reset the navigation history here?

Reply:
I initially reset nav history/index here, but I was getting unwanted behavior that `index.js` was added to navigation history twice. So I added the reset of history below at line 145.